addons/packages/app-toolkit/0.2.0/bundle/config/developer-namespace.yaml

```diff
@@ -0,0 +1,48 @@
+#@ load("@ytt:data", "data")
+#@ load("@ytt:yaml", "yaml")
+
+#@ if/end data.values.developer_namespace != "default":
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: #@ data.values.developer_namespace
+
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: registry-credentials
+  namespace: #@ data.values.developer_namespace
+  annotations:
+    secretgen.carvel.dev/image-pull-secret: ""
+type: kubernetes.io/dockerconfigjson
+data:
+  .dockerconfigjson: e30K
+
+---
+apiVersion: v1 
+kind: ServiceAccount
+metadata:
+  name: default
+  namespace: #@ data.values.developer_namespace
+  annotations:
+    kapp.k14s.io/create-strategy: fallback-on-update
+secrets:
+- name: registry-credentials
```

Comment:
Is this needed? I understand using a pull secret that may be used to load sidecar images from a private registry

Reply:
**Notes from Zoom Call:**
Having a separate `secret` and `imagePullSecret` for the supply chain to handle publishing of images and pulling of images is recommended. We should implement separate credentials rather than using the `registry-credentials` for both.

Add to the backlog for next release